# DATA PREPROCESSING
Exploring Mental Heath Dataset

In [53]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Kiểm tra tổng quan dữ liệu


In [54]:
df = pd.read_csv("./data/train.csv")
df.head(10)

id       Name  Gender   Age           City Working Professional or Student  \
0   0   Aaradhya  Female  49.0       Ludhiana            Working Professional   
1   1      Vivan    Male  26.0       Varanasi            Working Professional   
2   2     Yuvraj    Male  33.0  Visakhapatnam                         Student   
3   3     Yuvraj    Male  22.0         Mumbai            Working Professional   
4   4       Rhea  Female  30.0         Kanpur            Working Professional   
5   5       Vani  Female  59.0      Ahmedabad            Working Professional   
6   6     Ritvik    Male  47.0          Thane            Working Professional   
7   7    Rajveer    Male  38.0         Nashik            Working Professional   
8   8  Aishwarya  Female  24.0      Bangalore                         Student   
9   9     Simran  Female  42.0          Patna            Working Professional   

            Profession  Academic Pressure  Work Pressure  CGPA  \
0                 Chef                NaN            5.0   NaN   
1              Teacher                NaN            4.0   NaN   
2                  NaN                5.0            NaN  8.97   
3              Teacher                NaN            5.0   NaN   
4     Business Analyst                NaN            1.0   NaN   
5  Finanancial Analyst                NaN            2.0   NaN   
6              Chemist                NaN            5.0   NaN   
7              Teacher                NaN            3.0   NaN   
8                  NaN                2.0            NaN  5.90   
9          Electrician                NaN            4.0   NaN   

   Study Satisfaction  Job Satisfaction     Sleep Duration Dietary Habits  \
0                 NaN               2.0  More than 8 hours        Healthy   
1                 NaN               3.0  Less than 5 hours      Unhealthy   
2                 2.0               NaN          5-6 hours        Healthy   
3                 NaN               1.0  Less than 5 hours       Moderate   
4                 NaN               1.0          5-6 hours      Unhealthy   
5                 NaN               5.0          5-6 hours        Healthy   
6                 NaN               2.0          7-8 hours       Moderate   
7                 NaN               4.0          7-8 hours      Unhealthy   
8                 5.0               NaN          5-6 hours       Moderate   
9                 NaN               1.0          5-6 hours        Healthy   

    Degree Have you ever had suicidal thoughts ?  Work/Study Hours  \
0      BHM                                    No               1.0   
1      LLB                                   Yes               7.0   
2  B.Pharm                                   Yes               3.0   
3      BBA                                   Yes              10.0   
4      BBA                                   Yes               9.0   
5      MCA                                    No               7.0   
6       MD                                    No               6.0   
7  B.Pharm                                    No              10.0   
8      BSc                                    No               3.0   
9       ME                                   Yes               7.0   

   Financial Stress Family History of Mental Illness  Depression  
0               2.0                               No           0  
1               3.0                               No           1  
2               1.0                               No           1  
3               1.0                              Yes           1  
4               4.0                              Yes           0  
5               5.0                               No           0  
6               2.0                               No           0  
7               3.0                              Yes           0  
8               2.0                              Yes           0  
9               2.0                              Yes           0

In [55]:
df.shape

(140700, 20)

Kiểm tra dữ liệu bị thiếu

In [56]:
df.isnull().sum()

id                                            0
Name                                          0
Gender                                        0
Age                                           0
City                                          0
Working Professional or Student               0
Profession                                36630
Academic Pressure                        112803
Work Pressure                             27918
CGPA                                     112802
Study Satisfaction                       112803
Job Satisfaction                          27910
Sleep Duration                                0
Dietary Habits                                4
Degree                                        2
Have you ever had suicidal thoughts ?         0
Work/Study Hours                              0
Financial Stress                              4
Family History of Mental Illness              0
Depression                                    0
dtype: int64

# Nhận xét:

Ở các features Profession, Academic Pressure, Work Pressure, CGPA, Study Satisfaction, Job Satisfaction có số lượng NULL nhiều nhưng thật ra là vì giá trị các cột này phụ thuộc vào người đó là học sinh hay người đi làm
+ Phần lớn học sinh có đầy đủ giá trị các cột CGPA, Academic Pressure, Study Satisfaction.
+ Phần lớn người đi làm có đầy đủ giá trị các cột Professtion, Work Pressure, Job Satisfaction

Ngoài ra cũng có các cột Dietary Habits, Degree, Financial Stress thiếu một vài dữ liệu.

# Xử lý NULL

In [57]:
student_null_cgpa = df[(df['Working Professional or Student'] == 'Student') & (df['CGPA'].isnull())].shape[0]
print(f"Số lượng hàng là Student nhưng bị null CGPA: {student_null_cgpa}")

Số lượng hàng là Student nhưng bị null CGPA: 9


In [58]:
working_null_professional = df[(df['Working Professional or Student'] == 'Working Professional') & (df['Profession'].isnull())].shape[0]
print(f"Số lượng hàng là Working Professional nhưng bị null Profession: {working_null_professional}")

Số lượng hàng là Working Professional nhưng bị null Profession: 8763


In [59]:
null_pressure = df[df['Academic Pressure'].isnull() & df['Work Pressure'].isnull()].shape[0]
print(f"Số lượng hàng có giá trị null ở cả 2 cột Academic Pressure và Work Pressure: {null_pressure}")

Số lượng hàng có giá trị null ở cả 2 cột Academic Pressure và Work Pressure: 21


In [60]:
null_satisfaction = df[df['Study Satisfaction'].isnull() & df['Job Satisfaction'].isnull()].shape[0]
print(f"Số lượng hàng có giá trị null ở cả 2 cột Study Satisfaction và Job Satisfaction: {null_satisfaction}")

Số lượng hàng có giá trị null ở cả 2 cột Study Satisfaction và Job Satisfaction: 15


# Nhận xét: vì số lượng dữ liệu null của cả 2 cột Academic Pressure và Work Pressure quá nhỏ (15-21 mẫu) / dataset nên xử lý bằng cách loại bỏ, tương tự đối với 2 cột Study Satisfaction và Job Satisfaction.

In [61]:
df = df.dropna(subset=['Academic Pressure', 'Work Pressure'], how='all')

In [62]:
df = df.dropna(subset=['Study Satisfaction', 'Job Satisfaction'], how='all')

# Thực hiện loại bỏ null cho CGPA

In [63]:
df = df.drop(df[(df['Working Professional or Student'] == 'Student') & (df['CGPA'].isnull())].index)

# Thực hiện điền khuyết cho Profession

In [64]:
df.loc[(df['Working Professional or Student'] == 'Working Professional') & (df['Profession'].isnull()), 'Profession'] = 'no data'

# Ở các cột Dietary Habits, Degree, Financial Stress có vài dữ liệu thiếu nhưng nhìn chung vẫn rất ít (2-4 mẫu) / dataset, ta sẽ xử lý bằng cách loại bỏ.

In [65]:
df = df.dropna(subset=['Dietary Habits', 'Degree', 'Financial Stress'])

# Dataset sau khi xử lý null

In [66]:
df.isnull().sum()

id                                            0
Name                                          0
Gender                                        0
Age                                           0
City                                          0
Working Professional or Student               0
Profession                                27848
Academic Pressure                        112769
Work Pressure                             27887
CGPA                                     112769
Study Satisfaction                       112769
Job Satisfaction                          27885
Sleep Duration                                0
Dietary Habits                                0
Degree                                        0
Have you ever had suicidal thoughts ?         0
Work/Study Hours                              0
Financial Stress                              0
Family History of Mental Illness              0
Depression                                    0
dtype: int64

# Xử lý Outlies

# Cột Sleep duration

In [67]:
import re

# Chuẩn hóa về nhãn đúng và ánh xạ sang giá trị số
def normalize_and_map_sleep_duration(val):
    if not isinstance(val, str):
        return np.nan
    val = val.lower().strip()

    # Các trường hợp hợp lệ rõ ràng
    if val in ['less than 5 hours', '5-6 hours', '7-8 hours', 'more than 8 hours']:
        pass
    elif 'less' in val and '5' in val:
        val = 'less than 5 hours'
    elif '5' in val and '6' in val:
        val = '5-6 hours'
    elif '6' in val and '7' in val:
        val = '5-6 hours'
    elif '7' in val and '8' in val:
        val = '7-8 hours'
    elif 'more' in val or '8+' in val:
        val = 'more than 8 hours'
    else:
        # Nếu có số giờ lớn hơn 12 → loại bỏ
        hours = list(map(int, re.findall(r'\d+', val)))
        if hours and max(hours) > 12:
            return np.nan
        return np.nan  # không khớp mẫu

    # Ánh xạ sang giá trị số
    mapping = {
        'less than 5 hours': 4.5,
        '5-6 hours': 5.5,
        '7-8 hours': 7.5,
        'more than 8 hours': 8.5
    }
    return mapping.get(val, np.nan)

# Áp dụng
df['Sleep Duration'] = df['Sleep Duration'].apply(normalize_and_map_sleep_duration)

# Xem kết quả
print(df['Sleep Duration'].value_counts(dropna=False))

Sleep Duration
4.5    38769
7.5    36961
8.5    32718
5.5    32139
NaN       69
Name: count, dtype: int64


# Phương pháp:
# Chuyển đổi các giá trị có ý nghĩa tương đương sang các mốc 'Less than 5 hours', '7-8 hours', 'More than 8 hours', '5-6 hours' vì các mốc được liệt kê chiếm tần số xuất hiện lớn trong cột và bao quát được dữ liệu
# Những dữ liệu nhiễu không thuộc mốc nào thì chuyển đổi thành NaN để dễ xử lý hơn
# Chuẩn hóa 'Less than 5 hours' sang 4.5, '7-8 hours' sang 7.5, 'More than 8 hours' sang 8.5, '5-6 hours' sang 5.5 để dễ huấn luyện mô hình
# Sau khi chuyển đổi xong thì số lượng giá trị NaN không quá lớn so với tổng số dữ liệu trong cột

# Cột Dietary Habits

In [68]:
def clean_dietary(val):
    if not isinstance(val, str):
        return np.nan
    val = val.strip().lower()

    # Nhóm Moderate
    if (
        "moderate" in val
        or "less than healthy" in val
        or "less healthy" in val
    ):
        return "Moderate"

    # Nhóm Healthy
    if (
        "healthy" in val
        and not any(keyword in val for keyword in ["un", "no", "less"])
    ):
        return "Healthy"
    if val in ["more healthy"]:
        return "Healthy"

    # Nhóm Unhealthy
    if "unhealthy" in val or "no healthy" in val:
        return "Unhealthy"

    # Các giá trị không thuộc nhóm nào
    return np.nan

# Áp dụng làm sạch
df['Dietary Habits'] = df['Dietary Habits'].apply(clean_dietary)

# Kiểm tra kết quả
print(df['Dietary Habits'].value_counts(dropna=False))

Dietary Habits
Moderate     49688
Unhealthy    46216
Healthy      44734
NaN             18
Name: count, dtype: int64


# Phương pháp:
# Chuyển đổi các giá trị có ý nghĩa tương đương sang các giá trị 'Moderate', 'Unhealthy', 'Healthy' vì các giá trị được liệt kê chiếm tần số xuất hiện lớn trong cột và bao quát được dữ liệu
# Những dữ liệu nhiễu không thuộc giá trị chiếm tần số xuất hiện lớn nào thì chuyển đổi thành NaN để dễ xử lý hơn
# Sau khi chuyển đổi xong thì số lượng giá trị NaN không quá lớn so với tổng số dữ liệu trong cột

# Cột City

In [69]:
# Danh sách thành phố hợp lệ
sub_cities = [
    'Kalyan', 'Vasai-Virar', 'Patna', 'Kolkata', 'Meerut', 'Ahmedabad',
    'Rajkot', 'Ludhiana', 'Pune', 'Srinagar', 'Mumbai', 'Visakhapatnam',
    'Hyderabad', 'Agra', 'Vadodara', 'Indore', 'Surat', 'Varanasi',
    'Jaipur', 'Kanpur', 'Thane', 'Lucknow', 'Nagpur', 'Bangalore',
    'Chennai', 'Ghaziabad', 'Bhopal', 'Faridabad', 'Delhi', 'Nashik'
]

# Làm sạch cột City
df['City'] = df['City'].apply(lambda x: x if x in sub_cities else np.nan)

# Kiểm tra kết quả
print(df['City'].value_counts(dropna=False))

City
Kalyan           6590
Patna            5922
Vasai-Virar      5764
Kolkata          5688
Ahmedabad        5611
Meerut           5526
Ludhiana         5226
Pune             5207
Rajkot           5206
Visakhapatnam    5175
Srinagar         5072
Mumbai           4966
Indore           4871
Agra             4681
Surat            4635
Varanasi         4603
Vadodara         4564
Hyderabad        4493
Kanpur           4396
Jaipur           4328
Thane            4288
Lucknow          4279
Nagpur           4209
Bangalore        4122
Chennai          4043
Ghaziabad        3619
Delhi            3593
Bhopal           3474
Faridabad        3266
Nashik           3141
NaN                98
Name: count, dtype: int64


# Phương pháp:
# Vì sự khó khăn trong việc nhận dạng giá trị nào là tên thành phố, giá trị nào không phải nên cách hiệu quả để xử lý là xác định những giá trị tên thành phố xuất hiện nhiều trong cột(>=100)
# Sau đó những dữ liệu nhiễu không thuộc giá trị nào trong danh sách thành phố xuất hiện nhiều được lọc thì chuyển đổi thành NaN
# Sau khi chuyển đổi xong thì số lượng giá trị NaN không quá lớn so với tổng số dữ liệu trong cột

# Cột Profession

In [70]:
# Danh sách các nghề nghiệp hợp lệ
sub_professions = [
    "Teacher", "Content Writer", "Architect", "Consultant", "HR Manager", "Pharmacist",
    "Doctor", "Business Analyst", "Entrepreneur", "Chemist", "Chef", "Educational Consultant",
    "Data Scientist", "Researcher", "Lawyer", "Customer Support", "Marketing Manager", "Pilot",
    "Travel Consultant", "Plumber", "Sales Executive", "Manager", "Judge", "Electrician",
    "Financial Analyst", "Software Engineer", "Civil Engineer", "UX/UI Designer",
    "Digital Marketer", "Accountant", "Finanancial Analyst", "Mechanical Engineer",
    "Graphic Designer", "Research Analyst", "Investment Banker"
]

# Làm sạch cột Profession
df['Profession'] = df['Profession'].apply(lambda x: x if x in sub_professions else np.nan)

# Kiểm tra kết quả
print(df['Profession'].value_counts(dropna=False))

Profession
NaN                       36639
Teacher                   24906
Content Writer             7814
Architect                  4370
Consultant                 4228
HR Manager                 4022
Pharmacist                 3893
Doctor                     3255
Business Analyst           3161
Entrepreneur               2968
Chemist                    2967
Chef                       2862
Educational Consultant     2852
Data Scientist             2390
Researcher                 2328
Lawyer                     2212
Customer Support           2055
Marketing Manager          1976
Pilot                      1913
Travel Consultant          1860
Plumber                    1748
Sales Executive            1739
Manager                    1737
Judge                      1712
Electrician                1582
Financial Analyst          1574
Software Engineer          1510
Civil Engineer             1469
UX/UI Designer             1452
Digital Marketer           1372
Accountant                 13

# Phương pháp:
# Vì sự khó khăn trong việc nhận dạng giá trị nào là tên nghề nghiệp, giá trị nào không phải nên cách hiệu quả để xử lý là xác định những giá trị tên nghề nghiệp xuất hiện nhiều trong cột (>=100)
# Sau đó những dữ liệu nhiễu không thuộc giá trị nào trong danh sách nghề nghiệp xuất hiện nhiều được lọc thì chuyển đổi thành NaN
# Sau khi chuyển đổi xong thì số lượng giá trị NaN chiếm phần lớn so với tổng số dữ liệu trong cột

# Cột Degree

In [71]:
# Danh sách các giá trị hợp lệ (nghề nghiệp)
sub_degrees = [
    'Class 12','B.Ed','B.Arch','B.Com','B.Pharm','BCA','M.Ed','MCA','BBA','BSc','MSc','LLM','M.Pharm','M.Tech','B.Tech','LLB','BHM',
    'BA','ME','MD','MHM','BE','PhD','M.Com','MBBS','MA'
]

# Làm sạch cột Degree
df["Degree"] = df["Degree"].apply(lambda x: x if x in sub_degrees else np.nan)
# Kiểm tra số lượng
print(df["Degree"].value_counts(dropna=False))

Degree
Class 12    14704
B.Ed        11689
B.Arch       8741
B.Com        8112
B.Pharm      5855
BCA          5738
M.Ed         5667
MCA          5233
BBA          5029
BSc          5027
MSc          4878
LLM          4647
M.Pharm      4536
M.Tech       4475
B.Tech       4424
LLB          4348
BHM          4304
NaN          3931
BA           3750
ME           3631
MD           3383
MHM          3286
BE           3103
PhD          3103
M.Com        3094
MBBS         3081
MA           2887
Name: count, dtype: int64


# Phương pháp:
# Vì sự khó khăn trong việc nhận dạng giá trị nào là tên bằng, giá trị nào không phải cũng nên cách hiệu quả để xử lý là xác định những giá trị tên bằng xuất hiện nhiều trong cột (>=100)
# Sau đó những dữ liệu nhiễu không thuộc giá trị nào trong danh sách tên bằng xuất hiện nhiều được lọc thì chuyển đổi thành NaN
# Sau khi chuyển đổi xong thì số lượng giá trị NaN chiếm phần lớn so với tổng số dữ liệu trong cột

# Feature Engineering

Loại bỏ các cột không liên quan dựa đoán: id, Name

Tách target và data

In [72]:
X = df.drop(columns=["Depression", "id", "Name"])
y = df["Depression"]

Dựa trên dataset chủ yếu thuộc dạng categorical và các giá trị không liên tục, đồng thời chứa nhiều giá trị null khó có thể loại bỏ hay xử lí (vì các giá trị null này cũng mang 1 ý nghĩa riêng)

Các mô hình phù hợp nhất với dạng dữ liệu này là các mô hình có thể chấp nhận học với giá trị null:
- Mô hình dạng cây XGBoost, LightGBM, CatBoost, ...
- Mô hình tự xử lí missing value: TabNet,...

Tuy nhiên, nhằm tiếp cận dataset với các mô hình Binary Classification đơn giản: Logistic, Decision Tree

2 mô hình này cần xử lí Null

# Encode các cột dạng phân loại
Chuyển Yes/No thành True/False tức 0/1

In [73]:
X1 = X.copy()

In [74]:
X1['Working Professional or Student'] = X1['Working Professional or Student'].map({'Working Professional': True, 'Student': False})
X1['Gender'] = X1['Gender'].map({'Male': True, 'Female': False})

In [75]:
X1['Family History of Mental Illness'] = X1['Family History of Mental Illness'].map({'Yes': True, 'No': False})
X1['Have you ever had suicidal thoughts ?'] = X1['Have you ever had suicidal thoughts ?'].map({'Yes': True, 'No': False})

Dietary Habits chuyển thành -1 / 0 / 1

In [76]:
X1['Dietary Habits'] = X1['Dietary Habits'].map({'Unhealthy': -1, 'Moderate': 0, 'Healthy': 1})

In [77]:
X1

Gender   Age           City  Working Professional or Student  \
0        False  49.0       Ludhiana                             True   
1         True  26.0       Varanasi                             True   
2         True  33.0  Visakhapatnam                            False   
3         True  22.0         Mumbai                             True   
4        False  30.0         Kanpur                             True   
...        ...   ...            ...                              ...   
140695   False  18.0      Ahmedabad                             True   
140696   False  41.0      Hyderabad                             True   
140697   False  24.0        Kolkata                             True   
140698   False  49.0       Srinagar                             True   
140699    True  27.0          Patna                            False   

               Profession  Academic Pressure  Work Pressure  CGPA  \
0                    Chef                NaN            5.0   NaN   
1                 Teacher                NaN            4.0   NaN   
2                     NaN                5.0            NaN  8.97   
3                 Teacher                NaN            5.0   NaN   
4        Business Analyst                NaN            1.0   NaN   
...                   ...                ...            ...   ...   
140695                NaN                NaN            5.0   NaN   
140696     Content Writer                NaN            5.0   NaN   
140697  Marketing Manager                NaN            3.0   NaN   
140698            Plumber                NaN            5.0   NaN   
140699                NaN                4.0            NaN  9.24   

        Study Satisfaction  Job Satisfaction  Sleep Duration  Dietary Habits  \
0                      NaN               2.0             8.5             1.0   
1                      NaN               3.0             4.5            -1.0   
2                      2.0               NaN             5.5             1.0   
3                      NaN               1.0             4.5             0.0   
4                      NaN               1.0             5.5            -1.0   
...                    ...               ...             ...             ...   
140695                 NaN               4.0             5.5            -1.0   
140696                 NaN               4.0             7.5             0.0   
140697                 NaN               1.0             8.5             0.0   
140698                 NaN               2.0             5.5             0.0   
140699                 1.0               NaN             4.5             1.0   

          Degree  Have you ever had suicidal thoughts ?  Work/Study Hours  \
0            BHM                                  False               1.0   
1            LLB                                   True               7.0   
2        B.Pharm                                   True               3.0   
3            BBA                                   True              10.0   
4            BBA                                   True               9.0   
...          ...                                    ...               ...   
140695  Class 12                                  False               2.0   
140696    B.Tech                                   True               6.0   
140697     B.Com                                  False               4.0   
140698        ME                                   True              10.0   
140699       BCA                                   True               2.0   

        Financial Stress  Family History of Mental Illness  
0                    2.0                             False  
1                    3.0                             False  
2                    1.0                             False  
3                    1.0                              True  
4                    4.0                              True  
...                  ...                               ...  
1

# Kết hợp các cột liên quan
Như phân tích lúc đầu, tùy theo người đó `đi học` hay `đi làm` mà các cột `'Academic Pressure', 'Work Pressure', 'Study Satisfaction', 'Job Satisfaction'` có giá trị hay không.

Dựa trên điểm chung `Stress` hay `Satisfaction`, ta kết hợp 2 cột với nhau thành `'Aca/Work Pressure','Aca/Work Satisfaction'`

In [78]:
X1['Aca/Work Pressure'] = X1['Academic Pressure'].fillna(X1['Work Pressure'])
X1['Aca/Work Satisfaction'] = X1['Study Satisfaction'].fillna(X1['Job Satisfaction'])
X1.drop(columns=['Academic Pressure', 'Work Pressure', 'Study Satisfaction', 'Job Satisfaction'], inplace=True)

# Nhận xét:
Vì các cột Profession, City, Degree bao gồm rất nhiều mục, nên việc encoding khá khó khăn:
- One-hot encoding: tạo quá nhiều cột có thể gây dư thừa và tăng tính phức tạp lên rất cao
- Ordinary encoding: các mục không có tính thứ tự rõ ràng và số lượng không đồng đều

Nên phục vụ 2 mô hình đơn giản Logistic và Decision Tree, việc loại bỏ có thể chấp nhận được.

In [79]:
X1 = X1.drop(columns=['City', 'Degree', 'Profession'])

Cột CGPA có số lượng NULL tới 80% (112802 / 140700) nên việc xử lí điền khuyết, hay drop null là khá rủi ro đến sự chính xác của mô hình.
\
Vì thế với mô hình cơ bản, ta tạm bỏ qua cột CGPA.

In [80]:
X1 = X1.drop(columns=['CGPA'])

In [81]:
X1

Gender   Age  Working Professional or Student  Sleep Duration  \
0        False  49.0                             True             8.5   
1         True  26.0                             True             4.5   
2         True  33.0                            False             5.5   
3         True  22.0                             True             4.5   
4        False  30.0                             True             5.5   
...        ...   ...                              ...             ...   
140695   False  18.0                             True             5.5   
140696   False  41.0                             True             7.5   
140697   False  24.0                             True             8.5   
140698   False  49.0                             True             5.5   
140699    True  27.0                            False             4.5   

        Dietary Habits  Have you ever had suicidal thoughts ?  \
0                  1.0                                  False   
1                 -1.0                                   True   
2                  1.0                                   True   
3                  0.0                                   True   
4                 -1.0                                   True   
...                ...                                    ...   
140695            -1.0                                  False   
140696             0.0                                   True   
140697             0.0                                  False   
140698             0.0                                   True   
140699             1.0                                   True   

        Work/Study Hours  Financial Stress  Family History of Mental Illness  \
0                    1.0               2.0                             False   
1                    7.0               3.0                             False   
2                    3.0               1.0                             False   
3                   10.0               1.0                              True   
4                    9.0               4.0                              True   
...                  ...               ...                               ...   
140695               2.0               4.0                              True   
140696               6.0               5.0                              True   
140697               4.0               4.0                             False   
140698              10.0               1.0                             False   
140699               2.0               3.0                              True   

        Aca/Work Pressure  Aca/Work Satisfaction  
0                     5.0                    2.0  
1                     4.0                    3.0  
2                     5.0                    2.0  
3                     5.0                    1.0  
4                     1.0                    1.0  
...                   ...                    ...  
140695                5.0                    4.0  
140696                5.0                    4.0  
140697                3.0                    1.0  
140698                5.0                    2.0  
140699                4.0                    1.0  

[140656 rows x 11 columns]

# Chuẩn hoá dữ liệu

In [82]:
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()
X1_normalized = pd.DataFrame(scaler.fit_transform(X1), columns=X1.columns)

print(X1_normalized.head())

     Gender       Age  Working Professional or Student  Sleep Duration  \
0 -1.106832  0.695152                         0.497230        1.291788   
1  0.903480 -1.162225                         0.497230       -1.226583   
2  0.903480 -0.596936                        -2.011141       -0.596991   
3  0.903480 -1.485247                         0.497230       -1.226583   
4 -1.106832 -0.839203                         0.497230       -0.596991   

   Dietary Habits  Have you ever had suicidal thoughts ?  Work/Study Hours  \
0        1.256724                              -0.988843         -1.363057   
1       -1.230514                               1.011283          0.193972   
2        1.256724                               1.011283         -0.844047   
3        0.013105                               1.011283          0.972486   
4       -1.230514                               1.011283          0.712981   

   Financial Stress  Family History of Mental Illness  Aca/Work Pressure  \
0         